src/classes/UTIL_DMLService_TEST.cls

```diff
@@ -0,0 +1,699 @@
+        Account acct = new Account(Name = 'DMLOptionsAccount');
+        insert acct;
+
+        List<Contact> contacts = new List<Contact> { new Contact(FirstName = 'Blue', LastName = 'Cloud', AccountId = acct.Id),
+                                                     new Contact(FirstName = 'Astro', LastName = 'Cloud', AccountId = acct.Id) };
+                         
+        Test.startTest();
+        UTIL_DMLService.insertRecords(contacts, false, new Database.DMLOptions());
+        Test.stopTest();
+
+        List<Contact> retrievedInsertedContacts = [SELECT Id, FirstName, LastName, AccountId 
+                                                   FROM Contact];
+                                                   
+        if (!retrievedInsertedContacts.isEmpty()) {
+            System.assertEquals(contacts[0].FirstName, retrievedInsertedContacts[0].FirstName);
+            System.assertEquals(contacts[1].FirstName, retrievedInsertedContacts[1].FirstName);
+        }
+
+        contacts[0].FirstName = 'Cloudy';
+        contacts[0].FirstName = 'Cody';
+        UTIL_DMLService.updateRecords(contacts, false, new Database.DMLOptions());
+
+        List<Contact> retrieveUpdatedContacts = [SELECT Id, FirstName, LastName, AccountId 
+                                                 FROM Contact];
+
+        if (!retrieveUpdatedContacts.isEmpty()) {
+            System.assertEquals(contacts[0].FirstName, retrieveUpdatedContacts[0].FirstName);
+            System.assertEquals(contacts[1].FirstName, retrieveUpdatedContacts[1].FirstName);
+        }
+    }
+    
+
+    /*********************************************************************************************************
+    * @description Tests the NPSP Setting Respect Duplicate Rule Settings for duplicate Accounts
+    *   set to false 
+    * verify:
+    *   
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRespectDuplicateRuleSettingsForAccounts() {
+        if (strTestOnly != '*' && strTestOnly != 'testRespectDuplicateRuleSettingsForAccounts') return;
+        
+        UTIL_CustomSettingsFacade.getErrorSettingsForTests(new Error_Settings__c(Respect_Duplicate_Rule_Settings__c = false));
+        Error_Settings__c errorSettings = UTIL_CustomSettingsFacade.getErrorSettings();
+        UTIL_DMLService.insertRecord(errorSettings, true);
+        
+        Account acct = new Account(Name='Blue Cloud', BillingCity='San Francisco');
+        UTIL_DMLService.insertRecord(acct);
+        List<Account> retrievedAccts = [SELECT Id, Name FROM Account];
+        if (!retrievedAccts.isEmpty()) {
+            System.assertEquals('Blue Cloud', retrievedAccts[0].Name);
+        }
+        
+        Account dupeAcct = new Account(Name='Blue Cloud', BillingCity='San Francisco');
+        Database.SaveResult sr = UTIL_DMLService.insertRecord(dupeAcct);
+        if (!sr.isSuccess()) {
```

Comment:
i assume there is no way to create/turn on duplicate alerts in a test, and so the best we can do is see if it failed or not?

Reply:
Yes, that's correct.